packages/zoe/test/unitTests/contracts/test-autoswap.js

```diff
@@ -220,113 +562,305 @@ test('autoSwap - test fee', async t => {
+    // Now add to the liquidity pool in an exact ratio. (The pool has
+    // 12100 Moola and 11000 simoleans)
+    const liqDetails2 = {
+      mAmount: moola(121),
+      sAmount: simoleans(110),
+    };
+    const addLiqExpect2 = scaleForAdd(poolState, { m: 121, s: 110 });
+    alice.addLiquidityAndCheck(t, poolState, liqDetails2, addLiqExpect2);
+    poolState = updatePoolState(poolState, addLiqExpect2);
+    t.end();
+  } catch (e) {
+    t.assert(false, e);
+    console.log(e);
+  }
+});
 
-    // Bob claims it
-    const bobExclInvitation = await invitationIssuer.claim(bobInvitation);
-    const bobInstance = await E(zoe).getInstance(bobExclInvitation);
-    const bobInstallation = await E(zoe).getInstallation(bobExclInvitation);
-    t.equals(bobInstallation, bobInstallation);
+test('autoSwap - trade attempt before init', async t => {
+  try {
+    const {
+      moolaIssuer,
+      simoleanIssuer,
+      moolaMint,
+      moola,
+      simoleans,
+      zoe,
+    } = setup();
+    const installation = await installationPFromSource(zoe, autoswap);
 
-    const bobAutoswap = E(zoe).getPublicFacet(bobInstance);
-    // Bob looks up the price of 1000 moola in simoleans
-    const simoleanAmounts = await E(bobAutoswap).getCurrentPrice(
-      moola(1000),
-      simoleans(0).brand,
+    const issuerKeywordRecord = harden({
+      Central: moolaIssuer,
+      Secondary: simoleanIssuer,
+    });
+    const { publicFacet } = await zoe.startInstance(
+      installation,
+      issuerKeywordRecord,
     );
-    t.deepEquals(simoleanAmounts, simoleans(906), `simoleans out`);
 
-    // Bob escrows
-    const bobMoolaForSimProposal = harden({
-      give: { In: moola(1000) },
+    const moolaPurse = moolaIssuer.makeEmptyPurse();
+    moolaPurse.deposit(moolaMint.mintPayment(moola(100)));
+
+    const inAmount = moola(100);
+    const proposal = harden({
       want: { Out: simoleans(0) },
+      give: { In: inAmount },
     });
-    const bobMoolaForSimPayments = harden({ In: bobMoolaPayment });
+    const payment = harden({ In: moolaPurse.withdraw(inAmount) });
 
-    // Bob swaps
-    const bobSeat = await zoe.offer(
-      bobExclInvitation,
-      bobMoolaForSimProposal,
-      bobMoolaForSimPayments,
+    const seat = await zoe.offer(
+      E(publicFacet).makeSwapInvitation(),
+      proposal,
+      payment,
+    );
+    t.rejects(
+      seat.getOfferResult(),
+      /Pool not initialized/,
+      'The pool has not been initialized',
     );
 
-    assertOfferResult(t, bobSeat, 'Swap successfully completed.');
+    const { In: mPayout, Out: sPayout } = await seat.getPayouts();
+    assertPayoutAmount(t, moolaIssuer, mPayout, moola(100));
+    assertPayoutAmount(t, simoleanIssuer, sPayout, simoleans(0));
+
+    const poolPost = await E(publicFacet).getPoolAllocation();
+    t.deepEquals({}, poolPost, `empty Pool still`);
+
+    seat
+      .getOfferResult()
+      .then(async () =>
```

Comment:
hmm, if seat.getOfferResult() rejects, how does this get called, when it's the resolves callback?

Reply:
This needs to be awaited, or otherwise gathered up